### DISCUSSION TITLE
add CommentBox

### DISCUSSION THREAD

**Discussion by rastafrange** [facebook/docusaurus#3754] [Q&A]
Hello, I was wondering how to add [commentbox.io module](https://commentbox.io/docs)? I'd like to add it in both the doc and blog part.

I added this to `docusaurus.config.js `

```
 scripts: [
    "../js/commentbox.js",
  ],
```
and I created a file static/js/commentbox.js but not sure what to add exactly in this

Thanks

**Comment by slorber** [comment]
Hi

You can probably swizzle a theme component to add the comment box to the place you want.

**Reply by gettoknowu2** [reply]
Hi @slorber , 

Could you please clarify? I'm having the same trouble as @rastafrange as I'm trying to add a [Replybox](https://getreplybox.com/docs/getting-started-universal-code). 

I'm a newby to Docusaurus with no experience of this swizzling feature. 

I tried to manually add the scripts at the end of the html file and a `<div id="replybox"></div>` at the end of an article. The comment box however only loads after refreshing the page. There must be a much smoother way to get it up and running. 

Your help would be of much value to me.

**Reply by rastafrange** [reply]
@gettoknowu2 
I succeeded in adding commentBox.io to the pages I wanted, here's how I did it:
I've added the react script in a file "CommentBoxIO.js" under 'pages/components/CommentBoxIO'.

Then in each of my .md files where I want to have comments, I added 'import PageWithComments from "@site/src/components/CommentBoxIO/CommentBoxIO.js" ' right after 'id, slug, title ...' and at the end of the .md file I added "<PageWithComments />"

However I am not 100% convinced by commentbox.io (cannot custom css, needs to create an account to comment) but at least it's free for my small website, unlike replybox :/

**Reply by gettoknowu2** [reply]
@rastafrange 

Thanks for your instructions. They are of great help! I myself gave CommentBox a good try.

I however get stuck at some non-descriptive errors, like this one...

```Docusaurus Node/SSR could not render static page with path=/components/CommentBoxIO/CommentBoxIO because of error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.```

**Reply by rastafrange** [reply]
@gettoknowu2 maybe you should come on the [discord channel ](https://discord.com/channels/398180168688074762/567414046073159701) so that I'll be able to help you more easily ;)

**Reply by slorber** [reply]
@gettoknowu2 hard to help you without a repro
